File: python/Monitor.py
from Komponen import Komponen
from PeripheralDevice import PeripheralDevice
class Monitor(Komponen, PeripheralDevice) :
    def __init__(self, sizeInch, resolution, refreshRate, merk, nama, interfaceType, isWireless):
        Komponen.__init__(self, merk, nama)
        PeripheralDevice.__init__(self, interfaceType, isWireless)
        
        self.__sizeInch = sizeInch
        self.__resolution = resolution
        self.__refreshRate = refreshRate
        
    def get_sizeInch(self):
        return self.__sizeInch
    def get_resolution(self):
        return self.__resolution
    def get_refreshRate(self):
        return self.__refreshRate
    
    # Setter methods
    def set_sizeInch(self, sizeInch):
        self.__sizeInch = sizeInch
    def set_resolution(self, resolution):
        self.__resolution = resolution
    def set_refreshRate(self, refreshRate):
        self.__refreshRate = refreshRate
        

File: cpp/Monitor.cpp
// Monitor.cpp (Multiple Inheritance)
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "PeripheralDevice.cpp"
using namespace std;

class Monitor : public Komponen, public PeripheralDevice {
private:
    int sizeInch;
    string resolution;
    int refreshRate;
    
public:
    Monitor() {}
    
    Monitor(int sizeInch, string resolution, int refreshRate, 
           string interfaceType, bool isWireless,
           string merk, string nama) : 
           Komponen(merk, nama), PeripheralDevice(interfaceType, isWireless) {
        this->sizeInch = sizeInch;
        this->resolution = resolution;
        this->refreshRate = refreshRate;
    }
    
    void setSizeInch(int sizeInch) {
        this->sizeInch = sizeInch;
    }
    
    void setResolution(string resolution) {
        this->resolution = resolution;
    }
    
    void setRefreshRate(int refreshRate) {
        this->refreshRate = refreshRate;
    }
    
    int getSizeInch() {
        return this->sizeInch;
    }
    
    string getResolution() {
        return this->resolution;
    }
    
    int getRefreshRate() {
        return this->refreshRate;
    }
    
    ~Monitor() {}
};
